swap_meet/vendor.py

```diff
@@ -1,2 +1,140 @@
+from swap_meet.item import Item 
+
+
 class Vendor:
-    pass
+    def __init__(self, inventory=None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+        self.item = Item()
+
+
+    def add(self, item):
+        """
+        Takes one parameter item, adds to 
+        inventory, returns item
+        """
+        self.inventory.append(item)
+        return item 
+
+
+    def remove(self, item):
+        """
+        Takes one parameter item, removes from
+        inventory, returns item
+        """
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item 
+        else:
+            return False 
+
+
+    def get_by_category(self, category):
+        """
+        Takes one parameter category,
+        populates list items of a certain 
+        category, returns items
+        """
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+
+
+    def swap_items(self, other, my_item, other_item):
+        """
+        Takes in three parameters other (vendor), my_item,
+        other_item; returns Boolean based on whether swap 
+        is possible: only if the items are in stock
+        """
+        if my_item not in self.inventory or other_item not in other.inventory:
+            return False
+        else:
+            self.inventory.remove(my_item)
+            other.inventory.append(my_item)
+            other.inventory.remove(other_item)
+            self.inventory.append(other_item)
```

Comment:
this works fine! we could also use the `add` and `remove` instance methods instead.

```suggestion
            self.remove(my_item)
            other.add(my_item)
            other.remove(other_item)
            self.add(other_item)
```